src/mbgl/gl/context.hpp

```diff
@@ -155,11 +166,14 @@ class Context : private util::noncopyable {
 
     void setDirtyState();
 
+    std::unique_ptr<extension::Debugging> debugging;
+    std::unique_ptr<extension::VertexArray> vertexArray;
+
     State<value::ActiveTexture> activeTexture;
     State<value::BindFramebuffer> bindFramebuffer;
     State<value::Viewport> viewport;
     std::array<State<value::BindTexture>, 2> texture;
-    State<value::BindVertexArray> vertexArrayObject;
+    State<value::BindVertexArray, const Context&> vertexArrayObject { *this };
```

Comment:
We capture the `Context` into the state object and have it pass it to the individual functions.